prov/sm2/src/sm2_av.c

```diff
@@ -245,15 +207,20 @@ int sm2_av_open(struct fid_domain *domain, struct fi_av_attr *attr, struct fid_a
 	}
 
 	if (attr->type == FI_AV_UNSPEC)
-		attr->type = FI_AV_TABLE;
+		attr->type = FI_AV_MAP;
+
+	if (attr->type != FI_AV_MAP) {
+		FI_WARN(&sm2_prov, FI_LOG_AV, "Only FI_AV_MAP Supported\n");
+		return -FI_EINVAL;
+	}
```

Comment:
No applications really support FI_AV_MAP and it's really restrictive for the application. FI_AV_TABLE is much more common and usable with applications and if you use the util_av to store the fi_addr you will easily support it with no performance loss.

Reply:
done